### PR TITLE
🧹 Refactoring configureOAppPeers

### PR DESCRIPTION
**Pull request by ravinagill15**
## In this PR:
- Refactored `configureOAppPeers` in `ua-devtools > src > oapp > config.ts` to use `createConfigureEdges` pattern and added more logging

### CODE REVIEW

File: packages/ua-devtools/src/oapp/config.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import {
     formatOmniPoint,
     createConfigureMultiple,
     createConfigureNodes,
+    createConfigureEdges,
 } from '@layerzerolabs/devtools'
```
Comment (ravinagill15) [inline packages/ua-devtools/src/oapp/config.ts:13]: **Local development issue**: When in the `ua-devtools` directory, this package (and a few others in the imports section) is unrecognized in this config file if I install `node_modules` with `pnpm i`. 

If I then delete `node_modules` in `ua-devtools` and use `npm i` to install, they are recognized. I run into this issue with `pnpm i` even after calling `nvm use`.

This _might_ be related to the issue resolved by [this PR](https://github.com/LayerZero-Labs/devtools/pull/631) where we needed to signal to the `Dockerfile` to use the `pnpm` version specified in the `package.json`...

```
ravina@Ravinas-MacBook-Pro ua-devtools % npm -v
9.5.1
ravina@Ravinas-MacBook-Pro ua-devtools % pnpm -v
8.14.0
```

File: packages/ua-devtools/src/oapp/config.ts [hunk 2]
```diff
@@ -75,10 +73,41 @@ export const configureOAppPeers: OAppConfigurator = async (graph, createSdk) =>
...
+            },
+            {
+                onStart: (
+                    logger,
+                    [
+                        {
+                            vector: { from, to },
+                        },
+                    ]
+                ) => logger.verbose(`Checking connection for ${formatOmniVector({ from, to })}`),
```
Comment (ravinagill15) [inline packages/ua-devtools/src/oapp/config.ts:85]: Where would these actually be printed? Is it in the terminal when wiring 2 contracts together using `npx hardhat lz:oapp:wire --oapp-config layerzero.config.ts` ? If so, I don't see any of these logs (even before I made my changes) -- perhaps there is a flag for a more verbose logging level similar to `forge`'s `-vvvv`?

File: packages/ua-devtools/src/oapp/config.ts [hunk 3]
```diff
@@ -57,15 +58,12 @@ export const configureOAppDelegates: OAppConfigurator = withOAppLogger(
     }
 )
 
-export const configureOAppPeers: OAppConfigurator = async (graph, createSdk) => {
-    const logger = createModuleLogger('OApp')
-
-    return flattenTransactions(
```
Comment (ravinagill15) [inline packages/ua-devtools/src/oapp/config.ts:63]: Unsure if I still need to `flattenTransactions` if using `createConfigureEdges`...will double check stargate repo to see how it done there

File: packages/ua-devtools/src/oapp/config.ts [hunk 4]
```diff
@@ -75,10 +73,41 @@ export const configureOAppPeers: OAppConfigurator = async (graph, createSdk) =>
...
                 logger.verbose(`Creating a connection ${formatOmniVector({ from, to })}`)
                 return [await sdk.setPeer(to.eid, to.address)]
-            })
+            },
+            {
+                onStart: (
+                    logger,
+                    [
+                        {
+                            vector: { from, to },
```
Comment (janjakubnanista) [inline packages/ua-devtools/src/oapp/config.ts:82]: Here you don't need to destructure and restructure - just do:

```typescript
onStart: (logger, [{ vector }]) => logger.verbose(`Checking connection ${formatOmniVector(vector)}`),
```

File: packages/ua-devtools/src/oapp/config.ts (hunk 2 above)
Comment (janjakubnanista) [reply packages/ua-devtools/src/oapp/config.ts:85]: The log levels for our hardhat tasks are specified using `--log-level` CLI flag so in your case:

```bash
npx hardhat lz:oapp:wire --oapp-config layerzero.config.ts --log-level verbose
```

If you do

```bash
npx hardhat lz:oapp:wire --help
```

it should give you all the possible values of `-log-level`

File: packages/ua-devtools/src/oapp/config.ts [hunk 5]
```diff
@@ -75,10 +73,41 @@ export const configureOAppPeers: OAppConfigurator = async (graph, createSdk) =>
...
+                        },
+                    ]
+                ) => logger.verbose(`Checking connection for ${formatOmniVector({ from, to })}`),
+                onSuccess: (
+                    logger,
+                    [
+                        {
+                            vector: { from, to },
```
Comment (janjakubnanista) [inline packages/ua-devtools/src/oapp/config.ts:90]: Same as above

File: packages/ua-devtools/src/oapp/config.ts [hunk 6]
```diff
@@ -75,10 +73,41 @@ export const configureOAppPeers: OAppConfigurator = async (graph, createSdk) =>
...
+                        },
+                    ]
+                ) => logger.verbose(`${printBoolean(true)} Created connection for ${formatOmniVector({ from, to })}`),
+                onError: (
+                    logger,
+                    [
+                        {
+                            vector: { from, to },
```
Comment (janjakubnanista) [inline packages/ua-devtools/src/oapp/config.ts:98]: Same as above

**Review by janjakubnanista — CHANGES_REQUESTED** [review]

File: packages/ua-devtools/src/oapp/config.ts (hunk 4 above)
Comment (ravinagill15) [reply packages/ua-devtools/src/oapp/config.ts]: Makes sense !

File: packages/ua-devtools/src/oapp/config.ts (hunk 2 above)
Comment (ravinagill15) [reply packages/ua-devtools/src/oapp/config.ts]: Oo this is exactly what I was looking for, thank you!

**Review by janjakubnanista — APPROVED** [review]

**Review by janjakubnanista — APPROVED** [review]

**Pull request merged**